libraries/stdlib/src/kotlin/collections/Collections.kt

```diff
@@ -217,6 +217,7 @@ public inline fun <C, R> C.ifEmpty(defaultValue: () -> R): R where C : Collectio
 @kotlin.internal.InlineOnly
 public inline fun <@kotlin.internal.OnlyInputTypes T> Collection<T>.containsAll(elements: Collection<T>): Boolean = this.containsAll(elements)
 
+@PublishedApi
```

Comment:
`optimizeReadOnlyList` was introduced after the release of Kotlin 1.0. We cannot make it public and reference it from use code (via inlining) while we support `-api-version 1.0` option.

Reply:
Ah, I didn't know that. So should I manually copy its implementation into the inline functions I added it to, or is there a better solution?

Reply:
One solution is to wait when we drop support for `-api-version 1.0`.

Another option is to make an inline function that will either call `optimizeReadOnlyList` or do nothing, depending on the api version specified during the compilation. This technique is used here for example: https://github.com/JetBrains/kotlin/blob/v1.3.40/libraries/stdlib/jvm/src/generated/_ArraysJvm.kt#L393

Reply:
The api version check sounds good, but `apiVersionIsAtLeast` is JVM-only, so it can't currently be used from these functions. I can make an `expect` declaration for the new inline function to have the JVM implementation do a version check, and have the other platforms do nothing. That doesn't feel like a great solution, though. Is there a better way forward?

Reply:
`expect`/`actual` is a valid option. Other platforms have `optimizeReadOnlyList` since they become stable so there's no need to use `apiVersionIsAtLeast` there at all, just call that function.